### PR TITLE
feat(mgmt-ui): Replace integration UI with provider + sync config UI

### PR DESCRIPTION
**Pull request by asdfryan**
- Implement create and update flow for sync config
- Fix small bug with Select component

## Test Plan

Tested locally
https://www.loom.com/share/9c6e758de44c489f8d3335f1410510f3
https://www.loom.com/share/e434fa7e28244537be1dd6083a4a38da
https://www.loom.com/share/aa483eae172740cf9847595ec0a8397c
## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: apps/mgmt-ui/src/components/TabPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Box } from '@mui/material';
+
+interface TabPanelProps extends React.HTMLAttributes<HTMLDivElement> {
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/TabPanel/index.tsx:3]: Refactored from multiple files that defined the same thing

File: apps/mgmt-ui/src/components/configuration/provider/ProviderDetailsPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+  if (isLoading) {
+    return <Spinner />;
+  }
+
+  return (
+    <Card>
+      <Stack direction="column" className="gap-4" sx={{ padding: '2rem' }}>
+        <Stack direction="row" className="items-center justify-between w-full">
+          <Stack direction="row" className="items-center justify-center gap-2">
```
Comment (tomkit) [inline apps/mgmt-ui/src/components/configuration/provider/ProviderDetailsPanel.tsx:97]: nit: Could the two row Stacks be merged?

File: apps/mgmt-ui/src/components/configuration/syncConfig/SyncConfigDetailsPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+      },
+    };
+    return await createSyncConfig(activeApplicationId, newSyncConfig);
+  };
+
+  return (
+    <Card>
+      <Stack direction="column" className="gap-4" sx={{ padding: '2rem' }}>
+        <Stack direction="row" className="items-center justify-between w-full">
+          <Stack direction="row" className="items-center justify-center gap-2">
```
Comment (tomkit) [inline apps/mgmt-ui/src/components/configuration/syncConfig/SyncConfigDetailsPanel.tsx:141]: nit: same

File: apps/mgmt-ui/src/pages/api/internal/sync-configs/update.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import { getApplicationIdScopedHeaders } from '@/utils/headers';
-import { UpdateIntegrationResponse } from '@supaglue/schemas/v1/mgmt';
+import { UpdateSyncConfigResponse } from '@supaglue/schemas/v2/mgmt';
```
Comment (tomkit) [inline apps/mgmt-ui/src/pages/api/internal/sync-configs/update.ts:2]: nit: are we using v2 for everything now?

**Review by tomkit — APPROVED** [review]

File: apps/mgmt-ui/src/pages/api/internal/sync-configs/update.ts (hunk 4 above)
Comment (asdfryan) [reply apps/mgmt-ui/src/pages/api/internal/sync-configs/update.ts:2]: Yes

**Pull request merged**